### ISSUE TITLE
Bug Report - lastscandate fails

### ISSUE THREAD

**Issue by DerekSikes** [ArctosDB/arctos#8848]
**Describe the bug**
A clear and concise description of the encountered error.

all the specimens in this loan were scanned back into their unit trays 6 days ago:

https://arctos.database.museum/loanItemReview.cfm?transaction_id=21143714

but their lastscandates remain unchanged

same problem for specimens in this loan:

https://arctos.database.museum/loanItemReview.cfm?transaction_id=21126175

except one specimen, which updated to 2025-06-27 as it should

**Screenshots**

**Data**
If this involves external data, attach the __actual__ data that caused the problem. Do not attach a transformation or subset. You may ZIP most formats to attach, or request a Box email address for very large files.

**Desktop (please complete the following information):**

**Additional context**

**Comment by dustymc** [comment]
This is not a bug after all, bringing it back out here for more visibility.

The function is set to ignore anything done by the database (probably because it does all sorts of maintenance things that get logged, Just In Case), but the database also does (or did, see below) things (like bulkload) on behalf of users.

That's probably less-relevant now because https://github.com/ArctosDB/dev/issues/87 has been completed, so most/all of those 'on behalf of the user...' tasks are now logged as the user.

There's of course a huge backlog of data that hasn't been touched by a person since Arctos (maybe on behalf of some person) was there.

I don't care how this works and it's easy to change, but I don't want to surprise anyone with weird data either. If anyone uses this for something that changing the filter might break, please let me know ASAP - otherwise I'll just do whatever @DerekSikes tells me to do.

Here's the full history of an example container:

```
 select username,to_char(max(change_date),'YYYY-MM-DD') from container_history where container_id=13441174 group by username;;
   username   |  to_char   
--------------+------------
              | 
 arctosprod   | 2025-06-27
 jrykken      | 2024-10-24
 pereiratpl   | 2023-05-02
 JBWILLIAMSON | 2017-04-13
 MCMCHUGH     | 2014-07-24
```

and the "get last" function is executing and finding....

```

arctosprod@arctos>>  select to_char(max(change_date),'YYYY-MM-DD') from container_history where container_id=13441174 and username !='arctosprod';
  to_char   
------------
 2024-10-24
```

rather than the unfiltered view, which I think it probably what's wanted for this particular use case....

```


arctosprod@arctos>>  select to_char(max(change_date),'YYYY-MM-DD') from container_history where container_id=13441174;
  to_char   
------------
 2025-06-27

```

HELP!

**Comment by DerekSikes** [comment]
I don't really understand what you explain above.

I just want the lastscanneddate to be updated whenever the container the part is in is changed. Can you make that happen?

**Comment by dustymc** [comment]
>whenever the container the part is in is changed

Do you mean the part is placed into a new container (don't think so), or when something (including parentage) about the part-holding container changes (I think so).

**Comment by DerekSikes** [comment]
Whenever the part is put in a new container - the container that is the immediate parent of the part changes, then the date should update.

I would not want that date to update if, say we moved a bunch of drawers to new cabinets, since the parts are in unit trays (their immediate parent) inside the drawers and if those do not change then the date should not change. I really thought this was clear back when this happened: https://github.com/ArctosDB/arctos/issues/5644

**Comment by campmlc** [comment]
> Whenever the part is put in a new container - the container that is the immediate parent of the part changes, then the date should update.
> 
> I would not want that date to update if, say we moved a bunch of drawers to new cabinets, since the parts are in unit trays (their immediate parent) inside the drawers and if those do not change then the date should not change. I really thought this was clear back when this happened: [#5644](https://github.com/ArctosDB/arctos/issues/5644)

Agree with @DerekSikes 
This was my understanding as well. The date should reflect a change to the immediate parent container.

**Comment by DerekSikes** [comment]
is this because Arctos treats pins as containers? egad.... ? OK so change that... we want the NEXT higher parent of the part...

specimen (part) is on pin (container 1) - this almost never changes, but rarely we do swap out a barcode label on a pin for a new one (eg if the old one is damaged & unscannable).

the specimen+pin is inside a unit tray (container 2) and that is what changes frequently - whenever we move specimens for curation or send them on loan. But we also use this as a way of inventory by scanning all the specimens in a unit tray to tell Arctos "these are all present" - so no change has happened, but we want the lastscandate updated to indicate that these were seen & scanned.

Why not just have a question posed to the user whenever containers are scanned? "Do you want the lastscandate to be updated to today's date & time?"

**Comment by DerekSikes** [comment]
I would imagine the question would be a checkbox (or something) that only needed to be clicked once when using the tool (similar to the check box on this move container tool: https://arctos.database.museum/moveContainer.cfm )

but then we have to figure out how the user may specify what level of the container hierarchy triggers the lastscandate change...

**Comment by DerekSikes** [comment]
"Possibly displaying this one-shot summary in that specific form is simply confusing and we could - remove it? Replace that with the full history??? ???"

what would that look like? I am doubtful it would be possible if one is reviewing a list of perhaps thousands of parts and want to sort that list by a particular scan date for a particular container?

**Comment by DerekSikes** [comment]
Can we revive this? We would like to be able to see when we have scanned specimens back from a loan.

**Comment by dustymc** [comment]
I do not understand this request, closing as inactionable
